DuckDuckGoTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift

```diff
@@ -21,26 +21,919 @@ import XCTest
+        XCTAssertEqual(feature.transactionError, .failedToGetSubscriptionOptions)
+    }
+
+    func testGetSubscriptionOptionsReturnsEmptyOptionsWhenPurchaseNotAllowed() async throws {
+        // Given
+        let mockDependencyProvider = MockDependencyProvider()
+        mockDependencyProvider.subscriptionFeatureAvailability = SubscriptionFeatureAvailabilityMock(isFeatureAvailable: true,
+                                                                                                     isSubscriptionPurchaseAllowed: false,
+                                                                                                     usesUnifiedFeedbackForm: true)
+        AppDependencyProvider.shared = mockDependencyProvider
+
+        storePurchaseManager.subscriptionOptionsResult = Constants.subscriptionOptions
+
+        // When
+        let result = await feature.getSubscriptionOptions(params: Constants.mockParams, original: Constants.mockScriptMessage)
+
+        // Then
+        let subscriptionOptionsResult = try XCTUnwrap(result as? SubscriptionOptions)
+        XCTAssertEqual(subscriptionOptionsResult, SubscriptionOptions.empty)
+
+        XCTAssertEqual(feature.transactionStatus, .idle)
+        XCTAssertEqual(feature.transactionError, nil)
+    }
+
+    // MARK: - Tests for subscriptionSelected
+
+    func testSubscriptionSelectedSuccessWhenPurchasingFirstTime() async throws {
+        // Given
+        ensureUserUnauthenticatedState()
+
+        XCTAssertFalse(accountManager.isUserAuthenticated)
+
+        storePurchaseManager.hasActiveSubscriptionResult = false
+        storePurchaseManager.mostRecentTransactionResult = nil
+
+        authService.createAccountResult = .success(CreateAccountResponse(authToken: Constants.authToken,
+                                                                         externalID: Constants.externalID,
+                                                                         status: "created"))
+        authService.getAccessTokenResult = .success(AccessTokenResponse(accessToken: Constants.accessToken))
+        authService.validateTokenResult = .success(Constants.validateTokenResponse)
+        storePurchaseManager.purchaseSubscriptionResult = .success(Constants.mostRecentTransactionJWS)
+        subscriptionService.confirmPurchaseResult = .success(ConfirmPurchaseResponse(email: Constants.email,
+                                                                                     entitlements: Constants.entitlements,
+                                                                                     subscription: SubscriptionMockFactory.subscription))
+
+        // When
+        let subscriptionSelectedParams = ["id": "some-subscription-id"]
+        let result = await feature.subscriptionSelected(params: subscriptionSelectedParams, original: Constants.mockScriptMessage)
+
+        // Then
+        // TODO: Check pixel fired: DailyPixel.fireDailyAndCount(pixel: .privacyProPurchaseAttempt)
```

Comment:
Assuming these TODOs are just temporary while you review pixels.

Reply:
Correct, they will be gone as I do the pass for checking pixels.